app/controllers/attachments_controller.rb

```diff
@@ -119,4 +121,14 @@ def attachment_params
   def attachment_valid?
     !params["attachment"].nil? && !params["attachment"]["file"].nil?
   end
+
+  def ensure_extension_white_listed
```

Comment:
This seems to repeat some of the checking done in `attachment_valid?` and I wonder if it would be easier to have all checks done within that method and not use a before filter (sorry I know this was my original suggestion).
You could probably condense the logic down to something like:

```ruby
def attachment_valid?
  attachment = params["attachment"]
  attachment.present? && attachment["file"].present? && Attachment.valid_filetype(attachment["file"])
end
```